test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -294,6 +303,27 @@ class ExtAuthzHttpIntegrationTest : public HttpIntegrationTest,
     initiateClientConnection();
     waitForExtAuthzRequest();
 
+    AssertionResult result =
+        fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_);
+    RELEASE_ASSERT(result, result.message());
+    result = upstream_request_->waitForEndStream(*dispatcher_);
+    RELEASE_ASSERT(result, result.message());
+
+    // The original client request header value of "baz" is "foo". Since we configure to "override"
+    // the value of "baz", we expect the request headers to be sent to upstream contain only one
+    // "baz" with value "baz" (set by the authorization server).
+    EXPECT_THAT(upstream_request_->headers(), Http::HeaderValueOf("baz", "baz"));
+
+    // The original client request header value of "bat" is "foo". Since we configure to "append"
+    // the value of "bat", we expect the request headers to be sent to upstream contain two "bat"s,
+    // with values: "foo" and "bar" (the "bat: bar" header is appended by the authorization server).
+    const auto& baz_values =
+        TestUtility::getAllHeadersValuesForKey(upstream_request_->headers(), "bat");
+    EXPECT_TRUE(std::is_permutation(baz_values.begin(), baz_values.end(),
+                                    (std::vector<absl::string_view>{"foo", "bar"}).begin()));
+
```

Comment:
Can we add a test on appending multiple headers but those to be appended headers didn't exist in the original request?